### PR TITLE
Git Submodule issue #29

### PR DESCRIPTION
**Pull request by nelsonic**
Hi @SimonLab as described in #29 this PR tidies up the repo to have **`test-repo`** as a submodule. 📁 
Which should avoid the noisy `test msg` commits in the history. 🤞

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab please review when you can tomorrow. Thanks. 🙏

File: lib/httpoison_mock.ex [hunk 1]
```diff
@@ -148,21 +148,19 @@ defmodule Gitea.HTTPoisonMock do
...
-           status_code: 204
-         }}
-
-      true ->
-        {:ok,
-         %HTTPoison.Response{
-           body: Jason.encode!(%{deleted: List.first(String.split(url, "?"))}),
-           status_code: 200
-         }}
+    # match delete org
```
Comment (nelsonic) [inline lib/httpoison_mock.ex:151]: credo gave me a warning:

```sh
Cond statements should contain at least two conditions besides `true`, consider using `if` instead.
┃       lib/httpoison_mock.ex:151:5 #(Gitea.HTTPoisonMock.delete)
```

File: test/gitea_test.exs [hunk 2]
```diff
@@ -103,13 +96,18 @@ defmodule GiteaTest do
...
     git_repo_url = Gitea.Helpers.remote_url_ssh(org, repo)
 
-    assert {:ok, path} = Gitea.clone(git_repo_url)
+    # clone should work both in mock: true and mock: false
+    assert {:ok, _path} = Gitea.clone(git_repo_url)
     # assert path == Gitea.Helpers.local_repo_path(repo)
 
     # Clean up (but don't delete the remote repo!!)
-    delete_local_directory("public-repo")
+    Useful.empty_dir_contents(@git_dir)
```
Comment (nelsonic) [inline test/gitea_test.exs:110]: Figured this was a more fool-proof way of clearing out the `tmp/` dir ... https://github.com/dwyl/useful/pull/15 🤞

File: lib/git_mock.ex [hunk 3]
```diff
@@ -52,4 +58,33 @@ defmodule Gitea.GitMock do
...
   end
+
+  @doc """
+  `commit/2` (mock) simulate a commit and always returns {:ok, commit_info}
+
+    ## Examples
+    iex> GitMock.commit("my-repo", ["any", "args"])
+    {:ok, "[master dc5b0d4] test msg\n Author: Al Ex <c@t.co>\n 1 file changed, 1 insertion(+)\n"}
+  """
+  @spec commit(Git.Repository.t(), [any]) :: {:ok, any}
```
Comment (nelsonic) [inline lib/git_mock.ex:69]: need these for https://github.com/dwyl/gitea-demo/pull/5#issuecomment-1138454750 
ref: #28

**Review by SimonLab — APPROVED** [review]
LGTM,
I'll have a more detailed look at the tests soon also to see if we can improve some of them

**Comment by nelsonic** [comment]
Merci @SimonLab 🙌  hope your morning is going well so far. ☀️

**Pull request merged**